x/httpconnect/connect_client.go

```diff
@@ -16,96 +16,90 @@ package httpconnect
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"github.com/Jigsaw-Code/outline-sdk/transport"
 	"io"
 	"net"
 	"net/http"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
-// connectClient is a [transport.StreamDialer] implementation that dials [proxyAddr] with the given [dialer]
-// and sends a CONNECT request to the dialed proxy.
-type connectClient struct {
-	dialer    transport.StreamDialer
-	proxyAddr string
+// ConnectClient is a [transport.StreamDialer] that establishes an HTTP CONNECT tunnel over an abstract HTTP transport.
+//
+// The package also includes transport builders:
+// - NewHTTPProxyTransport
+// - NewHTTP3ProxyTransport
+//
+// Options:
+// - WithHTTPS to send "https://" CONNECT requests instead of "http://". Make sure transport is configured to use TLS.
+// - WithHeaders appends the provided headers to every CONNECT request.
+type ConnectClient struct {
+	transport http.RoundTripper
+	scheme    string
 
 	headers http.Header
 }
 
-var _ transport.StreamDialer = (*connectClient)(nil)
+var _ transport.StreamDialer = (*ConnectClient)(nil)
 
-type ClientOption func(c *connectClient)
+type ClientOption func(c *clientConfig)
 
-func NewConnectClient(dialer transport.StreamDialer, proxyAddr string, opts ...ClientOption) (transport.StreamDialer, error) {
-	if dialer == nil {
-		return nil, errors.New("dialer must not be nil")
-	}
-	_, _, err := net.SplitHostPort(proxyAddr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse proxy address %s: %w", proxyAddr, err)
+func NewConnectClient(transport http.RoundTripper, opts ...ClientOption) (*ConnectClient, error) {
```

Comment:
Let's make it clear that the proxy is encoded in the roundtripper.
```suggestion
func NewConnectClient(proxyRT http.RoundTripper, opts ...ClientOption) (*ConnectClient, error) {
```

Reply:
Done.